west.yml: Add hal gigadevice

nandojve opened the pull request:
This add modules/hal/gigadevice with three firmware libraries:

 - GD32E103 ARM (Cortex-M4F)
 - GD32F403 ARM (Cortex-M4F)
 - GD32VF103 RISC-V (Nucleisys Bumblebee core)

Signed-off-by: Gerson Fernando Budke <gerson.budke@atl-electronics.com>

gmarull commented on west.yml:
```diff
@@ -215,6 +217,11 @@ manifest:
       revision: c74be3890c9d975976fde1b1a3b2f5742bec34c0
       groups:
         - tee
+      revision: e18b7a9b040b5b5324520388047c9e7d678447e6
```
nit: sounds like this line should not be here

gmarull reviewed (changes requested):
Thanks for this effort, overall LGTM. I think that HAL should be compiled in a more efficient manner, as it is done, e.g. by STM32:

```cmake
# create a library and include sources as needed, e.g.
zephyr_sources(standard_peripheral/Source/gd32e10x_adc.c)
# should become
zephyr_library_sources_ifdef(CONFIG_USE_GD32_HAL_ADC standard_peripheral/Source/gd32e10x_adc.c)
```

nandojve commented:
rebase

carlescufi reviewed (approved):
Went through the current contents of the HAL repo, and they look fine.

carlescufi commented:
> But a `kconfig-ext: True` would be nice so that it is clear that the Kconfig resides in Zephyr repo.

@nandojve could you please add this to the HAL's `module.yml`?

tejlmand reviewed (commented):
Some minor nits observed that you can address if you like.

Also please reply to this:
https://github.com/zephyrproject-rtos/zephyr/pull/38658#discussion_r730852954

Could you update the commit message, currently it states:
>  west.yml: Add gigadevice hal
>
> This add modules/hal/gigadevice with three firmware libraries:
>
> - GD32E10X ARM (Cortex-M4F)
> - GD32F403 ARM (Cortex-M4F)
> - GD32VF103 RISC-V (Nucleisys Bumblebee core)

but with your latest push, this commit does a lot more than just updating `west.yml`

nandojve committed:
modules: Add gigadevice hal module

This add access to modules/hal/gigadevice with three firmware
libraries by updating west.yaml file:

 - GD32E10X ARM (Cortex-M4F)
 - GD32F403 ARM (Cortex-M4F)
 - GD32VF103 RISC-V (Nucleisys Bumblebee core)

It introduce module/hal_gigadevice, which contains all Cmake
and Kconfig rules to build GigaDevice SoCs into zephyr main
tree.

Signed-off-by: Gerson Fernando Budke <gerson.budke@atl-electronics.com>

nandojve commented:
Hi @tejlmand ,

Thank you for your comments.

> but with your latest push, this commit does a lot more than just updating `west.yml`

You are right, I split commit to distinguish both details. Since review started a lot of changes were made, including move rules from module to zephyr. If it is required, I can move rules commit to #38661.

tejlmand reviewed (changes requested):
Actually I was wrong on this, the PR is not bisectable.

If trying to bisect, the first commit fails as the glue code is not present with the new Zephyr module, so please squash the commits into a single commit..

Error seen:
```
Parsing /projects/github/ncs/zephyr/Kconfig
/projects/github/ncs/zephyr/scripts/kconfig/kconfig.py: build/Kconfig/Kconfig.modules:8: Could not open '/projects/github/ncs/zephyr/' (in 'osource "$(ZEPHYR_HAL_GIGADEVICE_KCONFIG)"') (EISDIR: Is a directory)
CMake Error at /projects/github/ncs/zephyr/cmake/kconfig.cmake:269 (message):
  command failed with return code: 1
Call Stack (most recent call first):
  /projects/github/ncs/zephyr/cmake/app/boilerplate.cmake:542 (include)
  /projects/github/ncs/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /projects/github/ncs/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:60 (include_boilerplate)
  CMakeLists.txt:5 (find_package)


-- Configuring incomplete, errors occurred!
```

nandojve commented:
> > I probably express myself wrongly, The error was just a compliance check, it not occurred at buildkite. When I told CI, I mean not 100% green on all checks.
> 
> yep, I see the error in the compliance check.
> 
> In that case, I would actually recommend that this PR becomes part of #38661 Of course it's always nice if PRs with multiple commits can be split into independent pieces, but it seems that this PR and #38661 are closely coupled.

After a lot of changes a more robust rules for this SoC were achieved. I afraid that if we join both PRs we may loose test conditions. For instance, The Fatal condition has been checked with/without CONFIG_SOC_SERIES_GD32xxx and passes on both cases. Since we are testing this PR and #38661 we cover both situations.
